### PR TITLE
[bitnami/mastodon]: support AUTHORIZED_FETCH and LIMITED_FEDERATION_MODE variables

### PR DESCRIPTION
**Pull request by dunn**
These variables are used to (1) require authentication for access to some data that is public by default, and (2) use an allow-list for federating servers.  The second mode requires the first.

- https://docs.joinmastodon.org/admin/config/#authorized_fetch
- https://docs.joinmastodon.org/admin/config/#limited_federation_mode

### CODE REVIEW

**Comment by carrodher** [comment]
Thank you for initiating this pull request. We appreciate your effort. Just a friendly reminder that it's important to sign your commits. Adding your signature certifies that you either authored the patch or have the necessary rights to contribute the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

Feel free to reach out if you have any questions or need assistance with the signing process.

**Comment by dunn** [comment]
thanks, added the sign-off, and added the variables to the readme while i was at it

**Comment by Mauraza** [comment]
Hi @dunn,

We usually use environment variables like `MASTODON_WEB_DOMAIN`.
Could you use this example and do the same with `AUTHORIZED_FETCH` and `LIMITED_FEDERATION_MODE`?

**Comment by dunn** [comment]
> Hi @dunn,
> 
> We usually use environment variables like `MASTODON_WEB_DOMAIN`. Could you use this example and do the same with `AUTHORIZED_FETCH` and `LIMITED_FEDERATION_MODE`?

Do you mean the user-facing variable should be prefixed with `MASTODON_` and translated to the original in `mastodon-env.sh`, or does the Bitnami version modify the variables across the board?

**Comment by Mauraza** [comment]
Hi @dunn,

like this:
https://github.com/bitnami/containers/blob/2de19b31fa4cc441d1e198e7f96818c9f7c0b535/bitnami/mastodon/4/debian-11/rootfs/opt/bitnami/scripts/mastodon-env.sh#L38
https://github.com/bitnami/containers/blob/2de19b31fa4cc441d1e198e7f96818c9f7c0b535/bitnami/mastodon/4/debian-11/rootfs/opt/bitnami/scripts/mastodon-env.sh#L75
https://github.com/bitnami/containers/blob/2de19b31fa4cc441d1e198e7f96818c9f7c0b535/bitnami/mastodon/4/debian-11/rootfs/opt/bitnami/scripts/mastodon-env.sh#L140-L142

In this form in the container, you can use `MASTODON_WEB_DOMAIN` or `WEB_DOMAIN`

**Comment by dunn** [comment]
done, thanks.

also for my own sanity i sorted the env var lists, if that's not something you'd like included we can drop it from the PR.

**Comment by Mauraza** [comment]
Hi @dunn,

We usually sort for types. If you check the file, we put first the environment variables starting with MASTODON... If you could only leave the file with the change of the new variables, that would be great. Sorry for the inconvenient

**Comment by dunn** [comment]
no problem, dropped that commit

File: bitnami/mastodon/4/debian-11/rootfs/opt/bitnami/scripts/mastodon-env.sh [hunk 1]
```diff
@@ -97,6 +97,8 @@ mastodon_env_vars=(
     AWS_SECRET_ACCESS_KEY
     AWS_ACCESS_KEY_ID
     S3_ENDPOINT
+    AUTHORIZED_FETCH
+    LIMITED_FEDERATION_MODE
```
Comment (Mauraza) [inline bitnami/mastodon/4/debian-11/rootfs/opt/bitnami/scripts/mastodon-env.sh:101]:
```suggestion
    MASTODON_AUTHORIZED_FETCH
    MASTODON_LIMITED_FEDERATION_MODE
    AUTHORIZED_FETCH
    LIMITED_FEDERATION_MODE
```

**Review by Mauraza — CHANGES_REQUESTED** [review]

**Review by Mauraza — DISMISSED** [review]
Hi @dunn,

Thanks a lot for the contribution!! this change should be available in the next revision.

**Review by Mauraza — APPROVED** [review]

**Pull request merged**